addons/mail/static/src/components/thread_view/thread_view_tests.js

```diff
@@ -649,6 +649,76 @@ QUnit.test('basic rendering of canceled notification', async function (assert) {
     );
 });
 
+QUnit.test('delete attachment should delete all text content free messages linked to it', async function (assert) {
+    assert.expect(3);
+
+    this.data['ir.attachment'].records.push({
+        id: 143,
+        mimetype: 'text/plain',
+        name: "Blah.txt",
+    });
+    this.data['mail.channel'].records.push({ id: 11 });
+    this.data['mail.message'].records.push(
+        {
+            attachment_ids: [143],
+            body: "hey man !",
+            channel_ids: [11],
+            id: 100,
+        }, {
+            attachment_ids: [143],
+            body: "",
+            channel_ids: [11],
+            id: 101,
+        }, {
+            attachment_ids: [143],
+            channel_ids: [11],
+            id: 102,
+        }
```

Comment:
It's very strange to link 3 messages to the same attachment, does that even happen in practice?

Reply:
It's `Many2many` on database so it should be possible, but I can't think of a flux. Why is it not a `one2many`?

Reply:
I think this test would be best split on 2: 1 test asserting the attachment delete is removing the message if empty, and 1 test asserting the message is kept if not empty
I don't think the case of 2 messages to 1 attachment is relevant.

Reply:
> It's `Many2many` on database so it should be possible, but I can't think of a flux. Why is it not a `one2many`?

I suppose it's to handle cases like "citation" or when a message is copied